packages/fleather/lib/src/widgets/editor_toolbar.dart

```diff
@@ -43,6 +45,80 @@ class InsertEmbedButton extends StatelessWidget {
   }
 }
 
+class UndoRedoButton extends StatelessWidget {
+  final FleatherController controller;
+  final _UndoRedoButtonVariant _variant;
+
+  const UndoRedoButton._(this.controller, this._variant, {Key? key})
+      : super(key: key);
+
+  const UndoRedoButton.undo({
+    Key? key,
+    required FleatherController controller,
+  }) : this._(controller, _UndoRedoButtonVariant.undo, key: key);
+
+  const UndoRedoButton.redo({
+    Key? key,
+    required FleatherController controller,
+  }) : this._(controller, _UndoRedoButtonVariant.redo, key: key);
+
+  bool _isEnabled() {
+    if (_variant == _UndoRedoButtonVariant.undo) {
+      return controller.history.canUndo;
+    } else {
+      return controller.history.canRedo;
+    }
+  }
+
+  void _onPressed() {
+    if (_variant == _UndoRedoButtonVariant.undo) {
+      _update(controller.history.undo());
+    } else {
+      _update(controller.history.redo());
+    }
+  }
+
+  void _update(Delta? changeDelta) {
+    if (changeDelta == null || changeDelta.isEmpty) {
+      return;
+    }
+
+    controller.compose(changeDelta,
+        selection: HistoryStack.selectionFromDelta(changeDelta),
+        source: ChangeSource.local);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+        animation: controller,
```

Comment:
What is the purpose of this? not sure I understand

Reply:
AnimatedBuilder rebuilds the widget when controller changes so that we can update whether button is active or not.